src/Components/ColorPicker/ColorPicker.tsx

```diff
@@ -60,35 +57,26 @@ export const ColorPicker = forwardRef<ColorPickerSwatchRef, ColorPickerProps>(
       colors = influxColors,
       testID = 'color-picker',
       maintainInputFocus = false,
+      validationFunc = validateHexCode,
     },
     ref
   ) => {
-    const [errorMessage, setErrorMessage] = useState('')
+    const errorMessage = validationFunc(color)
 
     const colorPickerClass = classnames('cf-color-picker', {
       [`${className}`]: className,
     })
 
-    const inputStatus = errorMessage
+    const inputStatus = !!errorMessage
       ? ComponentStatus.Error
       : ComponentStatus.Valid
 
     const handleSwatchClick = (hex: string): void => {
-      setErrorMessage('')
       onChange(hex, ComponentStatus.Valid)
     }
 
     const handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
-      const trimmedValue = e.target.value.trim()
-      const cleanedValue = trimmedValue
-        .split('')
-        .filter(char => HEX_CHARACTERS.includes(char.toLowerCase()))
-        .join('')
-
-      const newErrorMessage = validateHexCode(cleanedValue)
-
-      setErrorMessage(newErrorMessage)
-      onChange(cleanedValue, inputStatus)
+      onChange(e.target.value, inputStatus)
```

Comment:
why no more input cleaning here?

Reply:
I suppose I could trim out space characters at the very least